### PR TITLE
fix(coverage): correctly check semantic data after transform

### LINKED ISSUE #4999: transformer: incorrect semantic check
I saw nearly all tests have a `Symbols mismatch after transform` error. This error is thrown here

https://github.com/oxc-project/oxc/blob/081e2a37d96a765216f9bb0025208a9fdd699667/crates/oxc_semantic/src/post_transform_checker.rs#L129-L132

After looking into it, what happened here, I found a bug in `PostTransformChecker`. From the code above, it can be seen that this is due to the unequal length of semantic symbols in the two different phases. 

The `previous_collect` is collected in `after_semantic` and the `current_collect` is collected in `after_transform`. This is the cause of the problem. After the transformation, the `Semantic` will definitely change. 

So we shouldn't use `after_semantic`'s semantic to compare with `after_transform`'s semantic. Instead, we should rebuild the semantic after the transformation, and compare it to `after_transform`'s semantic

### PR DESCRIPTION
**Pull request by Boshen**
closes #4999

### CODE REVIEW

File: tasks/transform_conformance/babel.snap.md [hunk 1]
```diff
@@ -1275,97 +1276,88 @@ preset-env: unknown field `shippedProposals`, expected `targets` or `bugfixes`
...
+Missing ReferenceId
+Missing ReferenceId
+Missing ReferenceId
+Missing ReferenceId
+Missing ReferenceId
+Missing ReferenceId
+Missing ReferenceId
+Missing ReferenceId
+Missing ReferenceId
+Missing ReferenceId
```
Comment (Dunqing) [inline tasks/transform_conformance/babel.snap.md:1324]: What does happen here?

Comment (Boshen) [reply tasks/transform_conformance/babel.snap.md:1324]: let me check.

**Review by Dunqing — APPROVED** [review]

**Comment by overlookmotel** [comment]
I'm reading through this now. Please give me half an hour to finish review.

**Review by overlookmotel — APPROVED** [review]
This makes sense - it's now comparing post-transform scopes/symbols to a fresh semantic run on same AST.

There are further checks we can do to make sure the 2 sets of scopes/symbols in every way. I'll look into that in a follow-up.

**Pull request merged**